### DISCUSSION TITLE
is there expression syntax to do pattern matching (regex on strings)

### DISCUSSION THREAD

**Discussion by tombrus** [community/community#26186] [Actions]
<p>I would like to do a pattern match in ‘jobs.&lt;job_id&gt;.if’.<br>
Is that possible somehow?</p>
<p>e.g. if github.ref ~= feature/.*</p>
<p>-Tom</p>

**Comment by eddiechenms** [comment, marked as answer]
<p>Regex cannot be used in expression for now, as a workaround, you can use <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#functions" rel="nofollow noopener">Functions</a> to do some pattern matching.</p>
<p>For example, base on the information in your question: <span>if github.ref ~= feature/.*, you’d like to run the job when the trigger branch or tag is not started with “feature”, you can use “<strong>startsWith</strong>” to filter this:</span></p>
<pre><code>if: startWith(github.ref,'feature') != true
</code></pre>
<p>Hope this helps you.</p>

**Comment by huan** [comment]
<p>May I use the following expression?</p>
<pre><code>if: startsWith(github.ref, '/refs/heads/v[0-9]+')
</code></pre>

**Reply by rebeccafae** [reply]
Does anyone know the answer to this?

**Comment by psychemedia** [comment]
<p><code>startWith</code> or <code>startsWith</code> ? The answer suggests one then uses the other.</p>

**Reply by anz-ryu** [reply]
startsWith

**Comment by anz-ryu** [comment]
https://docs.github.com/en/actions/learn-github-actions/expressions#startswith
